compiler/optimizer/Structure.cpp

```diff
@@ -994,8 +994,19 @@ void TR_RegionStructure::cleanupAfterNodeRemoval()
       {
       TR_RegionStructure::Cursor si(*this);
       TR_ASSERT(si.getFirst() == getEntry(),"bad structure found trying to remove a block");
-      if (getParent())
-         getParent()->replacePart(this, getEntry()->getStructure());
+
+      auto parent = getParent();
+      if (!parent)
+         return;
+
+      auto structure = getEntry()->getStructure();
```

Comment:
I don't understand this logic. How can the entry node not have structure ? 

This routine in question is only responsible for updates to the structure representation, and so I am unclear on whether the null structure case is somehow meant to check that the entry is a "block structure" (as opposed to a TR::Block which is a CFG concept/construct). But if that is the intent I don't understand how check for null structure achieves that.

Reply:
Ah, I think I misunderstood the relationship between blocks and structures. I was under the impression that the entry node could not have structure, and thus possibly be NULL. I may have also confused `getStructure()` with `asStructure()`.

After your comment, I'm not actually sure there is a bug in this function. Instead, something up the call stack is invalidating the structure of an entry point. I'll mark this PR WIP and update it with information as the fix for that!